core/src/cluster_info.rs

```diff
@@ -2423,8 +2423,9 @@ impl ClusterInfo {
         self.stats
             .skip_push_message_shred_version
             .add_relaxed(num_crds_values - num_filtered_crds_values);
-        // Update crds values and obtain updated keys.
-        let updated_labels: Vec<_> = {
+        // Origins' pubkeys of updated crds values.
+        // TODO: Should this also include origins of new crds values?
+        let origins: HashSet<_> = {
```

Comment:
Hmmm, so first I'm assuming:

If a validator `V` pushes us a new crds value with origin `O`, then validator `V` gets added to the `received_set` for `O`.

So now the question is:
If there were earlier push messages with a *different* label, but for the same origin `O`, should this qualify `V` for pruning? Put another way, should the prune set be per origin, or per origin + label.

From my naive understanding, it should be per origin because the push decisions are all made based on the origin of the message, regardless of the label? So it seems like new crds values should also be included here, but it might be good to also double check with @aeyakovenko

Reply:
yeah, I agree with you